core/controller/transaction.ts

```diff
@@ -1,28 +1,26 @@
 import { ON } from 'core/util/decorator';
 import { BaseController } from 'core/controller/baseController';
 import { logger } from 'shared/util/logger';
-import { Transaction, TransactionModel, IAsset } from 'shared/model/transaction';
+import { IAsset, Transaction, TransactionModel } from 'shared/model/transaction';
 import TransactionQueue from 'core/service/transactionQueue';
 import TransactionService from 'core/service/transaction';
 import TransactionPool from 'core/service/transactionPool';
-import AccountRepository from 'core/repository/account';
 
 class TransactionController extends BaseController {
     @ON('TRANSACTION_RECEIVE')
     public async onReceiveTransaction(action: { data: { trs: TransactionModel<IAsset> } }): Promise<void> {
         const { data } = action;
         logger.debug(`[Controller][Transaction][onReceiveTransaction] ${JSON.stringify(data.trs)}`);
 
-        if (!TransactionService.validate(data.trs, AccountRepository.getByPublicKey(data.trs.senderPublicKey))) {
+        if (TransactionPool.has(data.trs)) {
```

Comment:
I think we need to validate transaction first

Reply:
validate more expensive operation then has. If id doesn't exist has will work correct

Reply:
but when trs will be `undefined`?